binding/Binding/SKObject.cs

```diff
@@ -1,29 +1,23 @@
 ﻿using System;
 using System.Collections.Concurrent;
-using System.Linq;
-using System.Reflection;
 using System.Runtime.InteropServices;
 using System.Threading;
 
 namespace SkiaSharp
 {
 	public abstract class SKObject : SKNativeObject
 	{
-#if THROW_OBJECT_EXCEPTIONS
-		internal static readonly ConcurrentBag<Exception> exceptions = new ConcurrentBag<Exception> ();
-#endif
+		private ConcurrentDictionary<IntPtr, SKObject> ownedObjects;
+		private ConcurrentDictionary<IntPtr, SKObject> keepAliveObjects;
 
-		internal static readonly ConcurrentDictionary<Type, ConstructorInfo> constructors;
-		internal static readonly ConcurrentDictionary<IntPtr, WeakReference> instances;
+		internal ConcurrentDictionary<IntPtr, SKObject> OwnedObjects =>
+			ownedObjects ??= new ConcurrentDictionary<IntPtr, SKObject> ();
```

Comment:
Lazy-instantiate this as it is hardly ever used. Typically for documents and some for some streams, codecs and typefaces.